src/components/TrapFocus/TrapFocus.tsx

```diff
@@ -84,9 +85,14 @@ export class TrapFocus extends React.PureComponent<TrapFocusProps, State> {
       event.preventDefault();
 
       if (event.srcElement === findFirstFocusableNode(focusTrapWrapper)) {
-        return focusLastFocusableNode(focusTrapWrapper);
+        return write(() => {
+          focusLastFocusableNode(focusTrapWrapper);
+        });
       }
-      focusFirstFocusableNode(focusTrapWrapper);
+
+      write(() => {
+        focusFirstFocusableNode(focusTrapWrapper);
+      });
```

Comment:
I had to wrap both of these in `write` to focus the button for all variations of using the mouse, escape button, and modal action item buttons.

Reply:
Is this for reflow or a delay? I.e, would [triggering a reflow](https://github.com/Shopify/polaris-react/blob/4199676447d8222560abbc766953c17c3ef92cd2/src/components/Frame/components/CSSAnimation/CSSAnimation.tsx#L78) work as well? Or is it for a delay? If it's a latter we might want to find a more reliable approach.

Reply:
This is for a delay. I don't think we need these `focusLastFocusableNode`/`first` calls with the modal since we're focusing the activator on our own, but I think these are called after we focus the activator, and so they change the focus to the body (I'm assuming because the modal has been removed and it can't find it?)

Reply:
Do we need this change at all?

Reply:
Yup, without it we lose focus of the activator

Reply:
Looks like v.5 is removing the app bridge logic from the modal so we probably won't even need to test in an app if we wait to merge this PR after that release.

I've also asked about removing `focusReturnPointNode` entirely from Modal since it doesn't appear to be used. https://github.com/Shopify/polaris-react/pull/2150#issuecomment-543297427